### ISSUE TITLE
[help wanted] LDA implemenation using tensorflow

### ISSUE THREAD

**Issue by mahatosourav91** [akashgit/Neural-Variational-Inference-for-Topic-Models#1]
Hi, I am trying to implement topic modelling using tensorflow.  I am trying to understand how to implemented it. What kind of preprocessing you did in the actual data?

**Comment by akashgit** [comment]
Pre-processing details are in the ICLR paper. But essentially, the documents are `vectorized` using `scikit`'s `CountVectorizer`

**Comment by mahatosourav91** [comment]
Thanks for the information. I have one more question, How can I get the weightage of each topics for each documents? By weightage I mean distribution of topics in each document. 
My goal is to cluster documents according to topics.

**Comment by akashgit** [comment]
thats quite straight forward and quick, simply change the `test()` method in the `model` file to output `tf.nn.softmax(self.z)`. This is the `\theta` or the topic proportions you are looking for.

**Comment by mahatosourav91** [comment]
Hi, sorry for bothering you again. Do you mean the following code give me the topic proportions :-
 theta = self.sess.run(tf.nn.softmax(self.z),feed_dict={self.x: np.expand_dims(X, axis=0),self.keep_prob: 1.0}) 
Is it correct?

If I do it like that way then I am getting an array of shape (100, 20)  for each document.[Please note that my batch_size = 100 and num_topics =20]. Is it possible to get an array of shape (1,20) for each document without altering my batch size.

**Comment by akashgit** [comment]
The above should work but it would be much easy if you simply ask the `test()` method to return `tf.nn.softmax(self.z)` along with the `cost` that it returning at the moment. The `test()` method accepts a single document at a time so it will also resolve your batch size problem.

**Comment by mahatosourav91** [comment]
Is it correct?
    `def test(self, X):`
        """Test the model and return the lowerbound on the log-likelihood.
        """
        `cost, theta = self.sess.run((self.cost, tf.nn.softmax(self.z)),feed_dict={self.x: np.expand_dims(X,` `axis=0),self.keep_prob: 1.0})`
        `return cost, theta`

**Comment by akashgit** [comment]
ya looks about right.

**Comment by honghongyier** [comment]
@mahatosourav91  Sorry to disturb you. I want to ask whether you fix " (batch_size, 50)" problem? I can't get the (1, 50) tensor as topic_prop for every document? Thanks for your attention.
